doc/examples/type_hints.rst

```diff
@@ -94,24 +94,34 @@ Typed Collection
 
 You can use :py:class:`~typing.TypedDict` (Python 3.8+) when using a well-defined schema for the data in a
 :class:`~pymongo.collection.Collection`. Note that all `schema validation`_ for inserts and updates is done on the server.
-These methods automatically add an "_id" field.
+These methods automatically add an "_id" field. In the example below the "_id" field is
+marked by the :py:class:`~typing_extensions.NotRequired` notation to allow it to be accessed when reading from
```

Comment:
I think the discussion of _id and NotRequired complicates what was a simple example and makes this section harder to read. What do you think about leaving this section as is and creating a new section to discuss modeling documents with TypedDict? There we could discuss these more advanced topics without complicating the simple intro example.

Reply:
I think you added the other example but did not revert changes to this one.